### PR TITLE
Add basic uniformity validation tests

### PR DESCRIPTION
**Pull request by alan-baker**
* Covers all collective operations
* Covers basic control flow
* Covers various expression types

Issue: #

<hr>

**Requirements for PR author:**

- [ ] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/uniformity/uniformity.spec.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    }
+    default: {
+      break;
+    }
+  }
+
+  return code;
+}
+
+export function generateOp({ op }: { op: string }) {
```
Comment (zoddicus) [inline src/webgpu/shader/validation/uniformity/uniformity.spec.ts:126]: As with `generateConditionalStatement` below, I am confused by the use of structured input here, and also I think a return type should be specified.

```
function generateOp(op: string) : string
```

File: src/webgpu/shader/validation/uniformity/uniformity.spec.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+  { cond: 'nonuniform_or1', expectation: false },
+  { cond: 'nonuniform_or2', expectation: false },
+  { cond: 'uniform_and', expectation: true },
+  { cond: 'nonuniform_and1', expectation: false },
+  { cond: 'nonuniform_and2', expectation: false },
+  { cond: 'uniform_func_var', expectation: true },
+  { cond: 'nonuniform_func_var', expectation: false },
+] as const;
+
+export function generateCondition({ condition }: { condition: string }) {
```
Comment (zoddicus) [inline src/webgpu/shader/validation/uniformity/uniformity.spec.ts:46]: As below, I am confused by the use of structured input here, and also I think a return type should be specified.

```
function generateCondition(condition: string): string
```

File: src/webgpu/shader/validation/uniformity/uniformity.spec.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  { op: 'dpdyFine', stage: 'fragment' },
+  { op: 'fwidth', stage: 'fragment' },
+  { op: 'fwidthCoarse', stage: 'fragment' },
+  { op: 'fwidthFine', stage: 'fragment' },
+  { op: 'storageBarrier', stage: 'compute' },
+  { op: 'workgroupBarrier', stage: 'compute' },
+  { op: 'workgroupUniformLoad', stage: 'compute' },
+] as const;
+
+export const kConditions = [
```
Comment (zoddicus) [inline src/webgpu/shader/validation/uniformity/uniformity.spec.ts:26]: Same comments for `kConditions` as `kCollectiveOps`

**Review by zoddicus — CHANGES_REQUESTED** [review]

File: src/webgpu/shader/validation/uniformity/uniformity.spec.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
+export const description = `Validation tests for uniformity analysis`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+export const kCollectiveOps = [
```
[the remark this answers is not in the document]
Comment (alan-baker) [reply src/webgpu/shader/validation/uniformity/uniformity.spec.ts]: Most of these boil down to me not knowing typescript and copying from somewhere else.

**Review by zoddicus — APPROVED** [review]

**Review by jrprice — APPROVED** [review]
This looks great! There's obviously lots of other stuff we should test as well, but I assume that we'll incrementally build up from here.

**Pull request merged**